### ISSUE TITLE
Multiple Editors, Show Popup focuses in last editor

### ISSUE THREAD

**Issue by SeanKendle** [froala/wysiwyg-editor#2744]
##### Expected behavior.
Showing the popup in an editor should focus in that editor, not the last editor on the page.

##### Actual behavior.
With multiple editors, if you show a popup, it focuses in the last editor. It is possible that this happens because I created a custom menu that is outside of Froala, and clicking that causes the current editor to lose focus. 

Once you click "OK" or "Cancel" in the Froala custom popup, (Firefox: it scrolls the page to the last editor, and) the last editor has focus.

##### Steps to reproduce the problem.
**froala_editor.js - Line 13333:**

```
function show (id, left, top, obj_height) {

      // Restore selection on show if it is there.
      if (!isVisible(id)) {
        if (areVisible() && editor.$el.find('.fr-marker').length > 0) {
          editor.events.disableBlur();
          editor.selection.restore();
        }
        else if (!areVisible()) {
	  console.info("editor", editor); //shows last editor on page
          // We must have focus into editor because we may want to save selection.
 
         //===> These lines cause the last editor on the page to gain focus:
          editor.events.disableBlur();
          editor.events.focus();
          editor.events.enableBlur();
        }

      }
```
##### OS.
Windows 7 Pro

##### Browser.
Firefox, browser scrolls
Chrome, browser doesn't scroll, but focus is in last editor
latest versions

**Comment by stefanneculai** [comment]
@SeanKendle Do you think it would be possible to make a jsFiddle to highlight the problem? That would speed-up things a lot. Here is a basic jsFiddle you could use as a starting point: https://jsfiddle.net/froala/wc5c3jhk/.

**Comment by SeanKendle** [comment]
I can say this, it didn't happen in v2.7.6, and it does now happen in v2.8.0.

I set up a jsfiddle.  It's not scrolling the page in this example, I don't know why, but it does seem to be focusing in the last editor on the page.
https://jsfiddle.net/SeanKendle/wc5c3jhk/342/

Watch the Console. I'm catching the "focus" event, and showing which editor is focused.

You have to:

1. Select some text to open the toolbar, Click on the 3d box looking icon for "widgets"
2. Insert a Blog Lead-in Widget, 
3. Select an "example" blog from the drop down. 
4. Click ok
5. Click on the new widget element
6. Click the broken-image button (should be a cog, not sure what's happening with that)
7. Click the edit button
8. Click Ok again
9. Look for "window.currentActiveFroala" in the console

**Comment by SeanKendle** [comment]
It feels like the popup's `data('instance')` is being set to the last created editor.

It seems to happen once this line runs:

```
// Restore selection on popups hide for the current active popup.
		editor.popups.onHide(id, function () {
			console.log("editor.popups.onHide()", $popup.data('instance'));
			var inst = $popup.data('instance') || editor;
        >>>> this line causes it:
        inst.accessibility.restoreSelection();
      })
```

**Comment by SeanKendle** [comment]
Here's what I've found. The `editor` variable supplied to the plugin cannot be relied upon throughout its lifecycle, as this variable gets set to the latest editor initialized.  So, instead, I'm storing a reference to the editor in the div on which I initialize the editor.

I show the hover menu on click for each of the modules in question, and so I then query for the parent `.fr-view` element and grab that data attribute which holds the editor.

This may not be the most efficient way to solve this problem, but it seems to work. I'm open to suggestions. Is there an easier/better way to get the current editor from within a plugin?

Here's my solution:

```
function _init() {
	$(document).off("click.module-jbl-form");
	$(document).on("click.module-jbl-form", "module-jbl[type='form']", function (e) {
		e.stopPropagation();
		$moduleForm = $(this);

		editor = $moduleForm.parents(".fr-view").data("froala-editor");

		$parentDiv = $moduleForm.parents("div");
		if ($parentDiv.length > 0) {
			let $parentMenu = $parentDiv.data("menu");
			if (typeof $parentMenu !== "undefined") {
				$parentDiv.removeClass("highlight");
				$parentMenu.hide();
			}
		}
		$(".highlight").removeClass("highlight");
		$moduleForm.addClass("highlight");

		let menu = getMenu($moduleForm);
		menu.data("hoverMenu").showMenu($moduleForm);
	});
}
```

**Comment by SeanKendle** [comment]
I think I've spotted the problem. It has entirely to do with my hover menu code and how it builds the menu on mouse over. I was resetting the on click listener, and setting it from the document scope. By that time the editor was referencing the last editor initialized on the page. I believe I have fixed this by setting the on click listener at the time of the plugin init function, and with the scope of editor.$el instead of $(document).

Thanks for your attention, I hope I didn't waste too much of your time.
